fixes #13, #7, #18, #3

notwaldorf opened the pull request:
- Fixes https://github.com/PolymerElements/paper-tooltip/issues/13 -- the animation was still happening on the second reshow, which makes the tooltip confused, so I cancel it for good measure
- Fixes https://github.com/PolymerElements/paper-tooltip/issues/7 -- `Polymer.dom(this).parentNode` is weird under `:root`. It gives the document fragment, not the actual node itself. I've played around with it in the shady and shadow dom, and I don't see `this.parentNode` giving the wrong result ever, though please shout if there's an obvious one I missed.
- Fixes https://github.com/PolymerElements/paper-tooltip/issues/18 and https://github.com/PolymerElements/paper-tooltip/issues/3 -- the `z-index` was in the wrong place.
- Fixes this not-working-at-all on IE10. Yeah, I just found out about how `hidden` is realllllllly not a thing.

/cc @morethanreal @cdata

cdata commented:
Re #7, under `:root` you can test to see if `parent.nodeType === Node.DOCUMENT_FRAGMENT`. This will allow you to tell if the parent you actually care about is the host.

Also, is it reasonable to test either of these bugfixes?

notwaldorf commented:
@cdata -- I've fixed the `parentNode` code again. It ended up much weirder than before, so I left a giant comment. Also, I've added tests to test the positioning for both plain and custom element parents, so, yay! :tada:

cdata commented on paper-tooltip.html:
```diff
@@ -140,16 +147,15 @@
+          target = Polymer.dom(ownerRoot).querySelector('#' + this.for);
         } else {
-          target = parentNode;
+          target = parentNode.nodeType == 11 ? ownerRoot.host : parentNode;
```
This is a little nit picky, but I think it's more sanitary to test against `Node.DOCUMENT_FRAGMENT_NODE` instead of the vague number `11`.

cdata commented:
One last issue, otherwise LGTM :+1:
